broken TestSlaveSyncRealRead unit test

Not sure if related to #310, but `TestSlaveSyncRealRead` seems to be broken on master, as it broke in a commit of the `adds-etcdconfig` feature branch.

Failure Snippet:

```
--- FAIL: TestSlaveSyncRealRead (20.65s)
	slavesync_test.go:54: == Start tlogserver with slave syncer== 
	slavesync_test.go:60: create slave syncer
	slavesync_test.go:61: create inmemory redis pool for ardb slave
	slavesync_test.go:75: create inmemory redis pool for tlog
	slavesync_test.go:81: start the server
	slavesync_test.go:85: listen addr = 127.0.0.1:42390
	slavesync_test.go:87: == start nbdserver backend with tlogrcp=127.0.0.1:42390 and slave pool
	slavesync_test.go:93: == generate some random data and write it to the nbdserver backend== 
	slavesync_test.go:137: flush data
	slavesync_test.go:145: CLOSING nbdstor
	slavesync_test.go:149: == validate all data are correct == 
	assertions.go:236: 
                          
	Error Trace:	slavesync_test.go:154
		
	            	
			slavesync_test.go:39
		
	Error:      	Expected nil, but got: &errors.errorString{s:"EOF"}
```

Full Log:

```
t=2017-07-14T11:06:56+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:46684" module=global caller=redis.go:49
t=2017-07-14T11:06:56+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:42062" module=global caller=redis.go:49
t=2017-07-14T11:06:56+0000 lvl=info msg="Started listening on local address 127.0.0.1:36875" module=global caller=server.go:56
t=2017-07-14T11:06:56+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:45393" module=global caller=redis.go:49
t=2017-07-14T11:06:56+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:45544" module=global caller=redis.go:49
t=2017-07-14T11:06:56+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-14T11:06:56+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:06:56+0000 lvl=dbug msg="subscribe 0xc420130420 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-14T11:06:56+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-14T11:06:56+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-14T11:06:56+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-14T11:06:56+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-14T11:06:56+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-14T11:06:56+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-14T11:06:56+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:06:56+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-14T11:06:56+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-14T11:06:56+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-14T11:06:56+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-14T11:06:56+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-14T11:06:56+0000 lvl=dbug msg="vdisk myimg's tlog server has received force flush message" module=global caller=tlog.go:339
t=2017-07-14T11:06:56+0000 lvl=info msg="slave syncer (myimg): killed" module=global caller=slave_sync.go:243
t=2017-07-14T11:06:56+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:248
t=2017-07-14T11:06:56+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:06:56+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-14T11:06:56+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:06:56+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-14T11:06:57+0000 lvl=dbug msg="no local content available for block 4 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:06:57+0000 lvl=dbug msg="no local content available for block 8 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:06:57+0000 lvl=dbug msg="no local content available for block 10 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:06:57+0000 lvl=dbug msg="no local content available for block 12 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:06:57+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:06:57+0000 lvl=dbug msg="unsubscribe 0xc420130420 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-14T11:06:57+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-14T11:06:57+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
t=2017-07-14T11:06:57+0000 lvl=info msg="connection from 127.0.0.1:47090 dropped" module=global caller=server.go:110
t=2017-07-14T11:06:57+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
t=2017-07-14T11:06:57+0000 lvl=dbug msg="flusher for tlogStorage myimg aborting" module=global caller=tlog.go:407
t=2017-07-14T11:06:57+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:217
t=2017-07-14T11:06:57+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:06:57+0000 lvl=dbug msg="background context for tlogStorage myimg aborting" module=global caller=tlog.go:352
t=2017-07-14T11:06:57+0000 lvl=info msg="GoBackground exited for vdisk: myimg" module=global caller=tlog.go:300
t=2017-07-14T11:06:57+0000 lvl=info msg="error while closing tlog client: close tcp 127.0.0.1:47090->127.0.0.1:36875: use of closed network connection" module=global caller=tlog.go:303
t=2017-07-14T11:06:57+0000 lvl=dbug msg="aborting HotReloader reloader" module=global caller=hotreload.go:201
t=2017-07-14T11:06:57+0000 lvl=dbug msg="aborting HotReloader listener" module=global caller=hotreload.go:157
t=2017-07-14T11:06:57+0000 lvl=dbug msg="transaction sender for tlogStorage myimg aborting" module=global caller=tlog.go:386
t=2017-07-14T11:06:57+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:40894" module=global caller=redis.go:49
t=2017-07-14T11:06:57+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:39067" module=global caller=redis.go:49
t=2017-07-14T11:06:57+0000 lvl=info msg="Started listening on local address 127.0.0.1:42390" module=global caller=server.go:56
t=2017-07-14T11:06:57+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:46849" module=global caller=redis.go:49
t=2017-07-14T11:06:57+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-14T11:06:57+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:06:57+0000 lvl=dbug msg="subscribe 0xc420245380 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-14T11:06:57+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:37330" module=global caller=redis.go:49
t=2017-07-14T11:06:57+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-14T11:06:57+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-14T11:06:57+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-14T11:06:57+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-14T11:06:57+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-14T11:06:57+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-14T11:06:57+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:06:57+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-14T11:06:57+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-14T11:06:57+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-14T11:06:57+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-14T11:06:57+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-14T11:06:57+0000 lvl=dbug msg="vdisk myimg's tlog server has received force flush message" module=global caller=tlog.go:339
t=2017-07-14T11:06:58+0000 lvl=info msg="slave syncer (myimg): killed" module=global caller=slave_sync.go:243
t=2017-07-14T11:06:58+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:248
t=2017-07-14T11:06:58+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:07:18+0000 lvl=eror msg="Failed to switchToArdbSlave: wait nbd slave sync timed out" module=global caller=tlog.go:217
t=2017-07-14T11:07:18+0000 lvl=eror msg="read failed:EOF" module=global caller=slavesync_test.go:155
t=2017-07-14T11:07:18+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:07:18+0000 lvl=dbug msg="unsubscribe 0xc420245380 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-14T11:07:18+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-14T11:07:18+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
--- FAIL: TestSlaveSyncRealRead (20.65s)
	slavesync_test.go:54: == Start tlogserver with slave syncer== 
	slavesync_test.go:60: create slave syncer
	slavesync_test.go:61: create inmemory redis pool for ardb slave
	slavesync_test.go:75: create inmemory redis pool for tlog
	slavesync_test.go:81: start the server
	slavesync_test.go:85: listen addr = 127.0.0.1:42390
	slavesync_test.go:87: == start nbdserver backend with tlogrcp=127.0.0.1:42390 and slave pool
	slavesync_test.go:93: == generate some random data and write it to the nbdserver backend== 
	slavesync_test.go:137: flush data
	slavesync_test.go:145: CLOSING nbdstor
	slavesync_test.go:149: == validate all data are correct == 
	assertions.go:236: 
                          
	Error Trace:	slavesync_test.go:154
		
	            	
			slavesync_test.go:39
		
	Error:      	Expected nil, but got: &errors.errorString{s:"EOF"}
t=2017-07-14T11:07:18+0000 lvl=info msg="Started listening on local address 127.0.0.1:40676" module=global caller=server.go:56
t=2017-07-14T11:07:18+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-14T11:07:18+0000 lvl=dbug msg="background context for tlogStorage myimg aborting" module=global caller=tlog.go:352
t=2017-07-14T11:07:18+0000 lvl=info msg="GoBackground exited for vdisk: myimg" module=global caller=tlog.go:300
t=2017-07-14T11:07:18+0000 lvl=info msg="error while closing tlog client: close tcp 127.0.0.1:42834->127.0.0.1:42390: use of closed network connection" module=global caller=tlog.go:303
t=2017-07-14T11:07:18+0000 lvl=dbug msg="flusher for tlogStorage myimg aborting" module=global caller=tlog.go:407
t=2017-07-14T11:07:18+0000 lvl=dbug msg="transaction sender for tlogStorage myimg aborting" module=global caller=tlog.go:386
t=2017-07-14T11:07:18+0000 lvl=dbug msg="aborting HotReloader reloader" module=global caller=hotreload.go:201
t=2017-07-14T11:07:18+0000 lvl=dbug msg="aborting HotReloader listener" module=global caller=hotreload.go:157
t=2017-07-14T11:07:18+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:44895" module=global caller=redis.go:49
t=2017-07-14T11:07:18+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:33745" module=global caller=redis.go:49
t=2017-07-14T11:07:18+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-14T11:07:18+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-14T11:07:18+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-14T11:07:18+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:42150" module=global caller=redis.go:49
t=2017-07-14T11:07:18+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-14T11:07:18+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-14T11:07:18+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-14T11:07:18+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:07:18+0000 lvl=dbug msg="subscribe 0xc4215fe540 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-14T11:07:18+0000 lvl=info msg="connection from 127.0.0.1:42834 dropped" module=global caller=server.go:110
t=2017-07-14T11:07:18+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
t=2017-07-14T11:07:18+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-14T11:07:18+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-14T11:07:18+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-14T11:07:18+0000 lvl=dbug msg="dynamicRedisPool listening" module=global caller=redispool.go:494
t=2017-07-14T11:07:18+0000 lvl=dbug msg="subscribe 0xc4215fe600 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-14T11:07:18+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-14T11:07:18+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-14T11:07:18+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-14T11:07:18+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-14T11:07:18+0000 lvl=dbug msg="vdisk myimg's tlog server has received force flush message" module=global caller=tlog.go:339
t=2017-07-14T11:07:18+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:07:18+0000 lvl=dbug msg="unsubscribe 0xc4215fe540 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-14T11:07:18+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-14T11:07:18+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
t=2017-07-14T11:07:18+0000 lvl=info msg="Started listening on local address 127.0.0.1:36484" module=global caller=server.go:56
t=2017-07-14T11:07:18+0000 lvl=info msg="connection from 127.0.0.1:49528 dropped" module=global caller=server.go:110
t=2017-07-14T11:07:18+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
t=2017-07-14T11:07:18+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-14T11:07:18+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-14T11:07:18+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-14T11:07:18+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-14T11:07:18+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:07:18+0000 lvl=dbug msg="subscribe 0xc422e97b60 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-14T11:07:18+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-14T11:07:18+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-14T11:07:18+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-14T11:07:18+0000 lvl=dbug msg="dynamicRedisPool listening" module=global caller=redispool.go:494
t=2017-07-14T11:07:18+0000 lvl=dbug msg="subscribe 0xc422e97c80 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-14T11:07:18+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-14T11:07:18+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:07:18+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-14T11:07:18+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-14T11:07:18+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-14T11:07:18+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-14T11:07:18+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-14T11:07:18+0000 lvl=info msg="slave syncer (myimg): killed" module=global caller=slave_sync.go:243
t=2017-07-14T11:07:18+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:248
t=2017-07-14T11:07:18+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:07:18+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-14T11:07:18+0000 lvl=dbug msg="reloadVdiskClusterConfigs exits early, no handlers registered" module=global caller=hotreload.go:230
t=2017-07-14T11:07:18+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-14T11:07:18+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-14T11:07:18+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-14T11:07:18+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-14T11:07:19+0000 lvl=dbug msg="no local content available for block 1 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:07:19+0000 lvl=dbug msg="no local content available for block 4 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:07:19+0000 lvl=dbug msg="no local content available for block 9 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-14T11:07:19+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-14T11:07:19+0000 lvl=dbug msg="unsubscribe 0xc422e97b60 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-14T11:07:19+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-14T11:07:19+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
FAIL
FAIL	github.com/zero-os/0-Disk/tlog/tlogserver	23.424s
```

This one really has to be fixed, as it now breaks any other commit quite regularly. Here is an excerpt of a log from the latest commit pushed to master:

```
t=2017-07-16T18:46:21+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:43438" module=global caller=redis.go:49
t=2017-07-16T18:46:21+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:35794" module=global caller=redis.go:49
t=2017-07-16T18:46:21+0000 lvl=info msg="Started listening on local address 127.0.0.1:37922" module=global caller=server.go:56
t=2017-07-16T18:46:21+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:41900" module=global caller=redis.go:49
t=2017-07-16T18:46:21+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:43157" module=global caller=redis.go:49
t=2017-07-16T18:46:21+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-16T18:46:21+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:21+0000 lvl=dbug msg="subscribe 0xc42001f140 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-16T18:46:21+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-16T18:46:21+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-16T18:46:21+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-16T18:46:21+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-16T18:46:21+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-16T18:46:21+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-16T18:46:21+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:21+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-16T18:46:21+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-16T18:46:21+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-16T18:46:21+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-16T18:46:21+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-16T18:46:21+0000 lvl=dbug msg="vdisk myimg's tlog server has received force flush message" module=global caller=tlog.go:339
t=2017-07-16T18:46:21+0000 lvl=info msg="slave syncer (myimg): killed" module=global caller=slave_sync.go:243
t=2017-07-16T18:46:21+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:248
t=2017-07-16T18:46:21+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:21+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-16T18:46:21+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:21+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-16T18:46:22+0000 lvl=dbug msg="no local content available for block 4 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:22+0000 lvl=dbug msg="no local content available for block 8 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:22+0000 lvl=dbug msg="no local content available for block 10 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:22+0000 lvl=dbug msg="no local content available for block 12 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:22+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:22+0000 lvl=dbug msg="unsubscribe 0xc42001f140 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-16T18:46:22+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-16T18:46:22+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
t=2017-07-16T18:46:22+0000 lvl=info msg="connection from 127.0.0.1:57484 dropped" module=global caller=server.go:110
t=2017-07-16T18:46:22+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
t=2017-07-16T18:46:22+0000 lvl=dbug msg="flusher for tlogStorage myimg aborting" module=global caller=tlog.go:407
t=2017-07-16T18:46:22+0000 lvl=dbug msg="background context for tlogStorage myimg aborting" module=global caller=tlog.go:352
t=2017-07-16T18:46:22+0000 lvl=info msg="GoBackground exited for vdisk: myimg" module=global caller=tlog.go:300
t=2017-07-16T18:46:22+0000 lvl=info msg="error while closing tlog client: close tcp 127.0.0.1:57484->127.0.0.1:37922: use of closed network connection" module=global caller=tlog.go:303
t=2017-07-16T18:46:22+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:217
t=2017-07-16T18:46:22+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:22+0000 lvl=dbug msg="aborting HotReloader reloader" module=global caller=hotreload.go:201
t=2017-07-16T18:46:22+0000 lvl=dbug msg="aborting HotReloader listener" module=global caller=hotreload.go:157
t=2017-07-16T18:46:22+0000 lvl=dbug msg="transaction sender for tlogStorage myimg aborting" module=global caller=tlog.go:386
t=2017-07-16T18:46:22+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:39170" module=global caller=redis.go:49
t=2017-07-16T18:46:22+0000 lvl=info msg="Started listening on local address 127.0.0.1:36086" module=global caller=server.go:56
t=2017-07-16T18:46:22+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-16T18:46:22+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:39426" module=global caller=redis.go:49
t=2017-07-16T18:46:22+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:44326" module=global caller=redis.go:49
t=2017-07-16T18:46:22+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:44063" module=global caller=redis.go:49
t=2017-07-16T18:46:22+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-16T18:46:22+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-16T18:46:22+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-16T18:46:22+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:22+0000 lvl=dbug msg="subscribe 0xc421b24960 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-16T18:46:22+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-16T18:46:22+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-16T18:46:22+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-16T18:46:22+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:22+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-16T18:46:22+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-16T18:46:22+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-16T18:46:22+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-16T18:46:22+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-16T18:46:22+0000 lvl=dbug msg="vdisk myimg's tlog server has received force flush message" module=global caller=tlog.go:339
t=2017-07-16T18:46:22+0000 lvl=info msg="slave syncer (myimg): killed" module=global caller=slave_sync.go:243
t=2017-07-16T18:46:22+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:248
t=2017-07-16T18:46:22+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:42+0000 lvl=eror msg="Failed to switchToArdbSlave: wait nbd slave sync timed out" module=global caller=tlog.go:217
t=2017-07-16T18:46:42+0000 lvl=eror msg="read failed:EOF" module=global caller=slavesync_test.go:155
t=2017-07-16T18:46:42+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:42+0000 lvl=dbug msg="unsubscribe 0xc421b24960 for vdisk myimg" module=global caller=hotreload.go:97
--- FAIL: TestSlaveSyncRealRead (20.65s)
	slavesync_test.go:54: == Start tlogserver with slave syncer== 
	slavesync_test.go:60: create slave syncer
	slavesync_test.go:61: create inmemory redis pool for ardb slave
	slavesync_test.go:75: create inmemory redis pool for tlog
	slavesync_test.go:81: start the server
	slavesync_test.go:85: listen addr = 127.0.0.1:36086
	slavesync_test.go:87: == start nbdserver backend with tlogrcp=127.0.0.1:36086 and slave pool
	slavesync_test.go:93: == generate some random data and write it to the nbdserver backend== 
	slavesync_test.go:137: flush data
	slavesync_test.go:145: CLOSING nbdstor
	slavesync_test.go:149: == validate all data are correct == 
	assertions.go:239: 
                          
	Error Trace:	slavesync_test.go:154
		
	            	
			slavesync_test.go:39
		
	Error:      	Expected nil, but got: &errors.errorString{s:"EOF"}
t=2017-07-16T18:46:42+0000 lvl=info msg="Started listening on local address 127.0.0.1:44198" module=global caller=server.go:56
t=2017-07-16T18:46:42+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-16T18:46:42+0000 lvl=dbug msg="background context for tlogStorage myimg aborting" module=global caller=tlog.go:352
t=2017-07-16T18:46:42+0000 lvl=info msg="GoBackground exited for vdisk: myimg" module=global caller=tlog.go:300
t=2017-07-16T18:46:42+0000 lvl=info msg="error while closing tlog client: close tcp 127.0.0.1:48022->127.0.0.1:36086: use of closed network connection" module=global caller=tlog.go:303
t=2017-07-16T18:46:42+0000 lvl=dbug msg="flusher for tlogStorage myimg aborting" module=global caller=tlog.go:407
t=2017-07-16T18:46:42+0000 lvl=dbug msg="transaction sender for tlogStorage myimg aborting" module=global caller=tlog.go:386
t=2017-07-16T18:46:42+0000 lvl=dbug msg="aborting HotReloader reloader" module=global caller=hotreload.go:201
t=2017-07-16T18:46:42+0000 lvl=dbug msg="aborting HotReloader listener" module=global caller=hotreload.go:157
t=2017-07-16T18:46:42+0000 lvl=info msg="connection from 127.0.0.1:48022 dropped" module=global caller=server.go:110
t=2017-07-16T18:46:42+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
t=2017-07-16T18:46:42+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:42182" module=global caller=redis.go:49
t=2017-07-16T18:46:42+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:39810" module=global caller=redis.go:49
t=2017-07-16T18:46:42+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-16T18:46:42+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-16T18:46:42+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-16T18:46:42+0000 lvl=info msg="embedded LedisDB Server ready and listening at 127.0.0.1:37698" module=global caller=redis.go:49
t=2017-07-16T18:46:42+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-16T18:46:42+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-16T18:46:42+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-16T18:46:42+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:42+0000 lvl=dbug msg="subscribe 0xc42023d8c0 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-16T18:46:42+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-16T18:46:42+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-16T18:46:42+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-16T18:46:42+0000 lvl=dbug msg="dynamicRedisPool listening" module=global caller=redispool.go:494
t=2017-07-16T18:46:42+0000 lvl=dbug msg="subscribe 0xc42023d980 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-16T18:46:42+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-16T18:46:42+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
t=2017-07-16T18:46:42+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-16T18:46:42+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-16T18:46:42+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-16T18:46:42+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-16T18:46:42+0000 lvl=dbug msg="vdisk myimg's tlog server has received force flush message" module=global caller=tlog.go:339
t=2017-07-16T18:46:43+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:43+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-16T18:46:43+0000 lvl=dbug msg="unsubscribe 0xc42023d8c0 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-16T18:46:43+0000 lvl=info msg="Started listening on local address 127.0.0.1:41231" module=global caller=server.go:56
t=2017-07-16T18:46:43+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
t=2017-07-16T18:46:43+0000 lvl=info msg="connection from 127.0.0.1:36196 dropped" module=global caller=server.go:110
t=2017-07-16T18:46:43+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
t=2017-07-16T18:46:43+0000 lvl=dbug msg="creating tlogStorage for backend myimg (db)" module=global caller=ardb.go:172
t=2017-07-16T18:46:43+0000 lvl=info msg="ready to reload config(s) upon SIGHUP receival" module=global caller=hotreload.go:123
t=2017-07-16T18:46:43+0000 lvl=dbug msg="sighupListener active" module=global caller=hotreload.go:131
t=2017-07-16T18:46:43+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:43+0000 lvl=dbug msg="subscribe 0xc42140af60 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-16T18:46:43+0000 lvl=dbug msg="defaultHotReloader reloader is active" module=global caller=hotreload.go:177
t=2017-07-16T18:46:43+0000 lvl=dbug msg="received handshake request from incoming connection" module=global caller=server.go:146
t=2017-07-16T18:46:43+0000 lvl=dbug msg="incoming connection checks out with version: 1.1.0-alpha" module=global caller=server.go:156
t=2017-07-16T18:46:43+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-16T18:46:43+0000 lvl=dbug msg="dynamicRedisPool listening" module=global caller=redispool.go:494
t=2017-07-16T18:46:43+0000 lvl=dbug msg="subscribe 0xc421f9cfc0 for vdisk myimg" module=global caller=hotreload.go:83
t=2017-07-16T18:46:43+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-16T18:46:43+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:43+0000 lvl=dbug msg="create vdisk with expectedSequence:0" module=global caller=vdisk_mgr.go:66
t=2017-07-16T18:46:43+0000 lvl=info msg="vdisk myimg reset first sequence to 0" module=global caller=vdisk.go:173
t=2017-07-16T18:46:43+0000 lvl=dbug msg="handshake phase successfully completed" module=global caller=server.go:181
t=2017-07-16T18:46:43+0000 lvl=dbug msg="replying handshake with status: OK" module=global caller=server.go:208
t=2017-07-16T18:46:43+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-16T18:46:43+0000 lvl=info msg="slave syncer (myimg): killed" module=global caller=slave_sync.go:243
t=2017-07-16T18:46:43+0000 lvl=info msg="slave syncer (myimg): exited" module=global caller=slave_sync.go:248
t=2017-07-16T18:46:43+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:43+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-16T18:46:43+0000 lvl=info msg="slavesync mgr: create for vdisk: myimg" module=global caller=slave_sync.go:58
t=2017-07-16T18:46:43+0000 lvl=dbug msg="redisProvider listening" module=global caller=ardb.go:324
t=2017-07-16T18:46:43+0000 lvl=dbug msg="reloadVdiskClusterConfigs exits early, no handlers registered" module=global caller=hotreload.go:230
t=2017-07-16T18:46:43+0000 lvl=dbug msg="(re)loading redis pools, using the latest available config" module=global caller=redispool.go:469
t=2017-07-16T18:46:43+0000 lvl=info msg="slave syncer (myimg): started" module=global caller=slave_sync.go:183
t=2017-07-16T18:46:44+0000 lvl=dbug msg="no local content available for block 1 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:44+0000 lvl=dbug msg="no local content available for block 4 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:44+0000 lvl=dbug msg="no local content available for block 9 and no remote storage available: no root ardb connection available" module=global caller=nondeduped.go:162
t=2017-07-16T18:46:44+0000 lvl=dbug msg="exit redisProvider listener" module=global caller=ardb.go:333
t=2017-07-16T18:46:44+0000 lvl=dbug msg="unsubscribe 0xc42140af60 for vdisk myimg" module=global caller=hotreload.go:97
t=2017-07-16T18:46:44+0000 lvl=info msg="tlogclient : reconnect from read because of : EOF" module=global caller=client.go:335
t=2017-07-16T18:46:44+0000 lvl=info msg="tlogclient: reconnect because `EOF` failed: client already closed" module=global caller=client.go:341
t=2017-07-16T18:46:44+0000 lvl=info msg="connection from 127.0.0.1:45410 dropped" module=global caller=server.go:110
t=2017-07-16T18:46:44+0000 lvl=dbug msg="abort current sendResp goroutine for vdisk:myimg" module=global caller=server.go:319
FAIL
FAIL	github.com/zero-os/0-Disk/tlog/tlogserver	23.393s
```

When a travis CI test fails it should be because that commit broke, not because of some unrelated undiscovered bug in the past, so this one is pretty annoying.

--- a/tlog/tlogclient/client.go
+++ b/tlog/tlogclient/client.go
@@ -443,14 +443,18 @@ func (c *Client) WaitNbdSlaveSync() error {
 // wait for a condition to happens
 // it happens when the channel is closed
 func (c *Client) waitCond(cond *sync.Cond) chan struct{} {
-	// create goroutine to listen to it's completeness
-	doneCh := make(chan struct{})
+	doneCh := make(chan struct{})        // channel to wait for waiter completion
+	waiterStarted := make(chan struct{}) // channel to wait for waiter start
+
 	go func() {
 		cond.L.Lock()
+		waiterStarted <- struct{}{}
 		cond.Wait()
 		cond.L.Unlock()
 		close(doneCh)
 	}()
+
+	<-waiterStarted
 	return doneCh
 }
 

